app/assets/javascripts/global_listeners.js

```diff
@@ -64,16 +74,27 @@ $(function() {
       };
     });
 
-  $(".menu-toggle").click(function(e) {
+  $("#sidebar-wrapper").mouseenter(function() {
+    var sidebar = $(this);
+    var timeoutId = setTimeout(function() {
+      sidebar.toggleClass("open");
+    }, 500);
+    sidebar.mouseleave(function() {
+      clearTimeout(timeoutId);
+      sidebar.removeClass("open");
+    });
+  });
+  
+  $("#sidebar-toggle").click(function(e) {
     e.preventDefault();
-    if($("#sidebar-wrapper").is(':hidden')) {
-      $('.click-overlay').on('click', function() {
-        hideSidebar();
-      });
-      showSidebar();
-    } else {
-      hideSidebar();
-    }
+    var wrapper = $("#sidebar-wrapper");
+    
+    if (wrapper.hasClass('collapsed') == true) 
+      $(this).html('<i class="mi md-18 mi--sidebar">close</i>');
```

Comment:
Can't we just change the text instead of changing the whole HTML all the time? I mean, so we don't have to have (duplicated) HTML on JS?